### PR TITLE
test(MS Teams plugin): bot messages for v1.10

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Document test cases for two classes of bot messages introduced with the MS Teams plugin v1.10.

#### Is the feature in stable branch or under review?
- [x] Feature in stable branch (master, main)? Yes, shipped in v1.10
- [x] Feature still under review? No, already merged.

#### Test server
https://msteams-sync-test.test.mattermost.cloud/

#### Test client
No specific requirements.

- [x] Browser
- [x] Mobile App (version, OS)
- [x] Desktop App (version, OS)

### CODE REVIEW

**Comment by lindy65** [comment]
Hey @lieut-data - this is awesome! Thank you for adding these tests :)

One comment...  In all the tests except for the last one, the verification step at the end of each test says `Very` instead of `Verify`

Otherwise looks good to me!

**Review by lindalumitchell — APPROVED** [review]
Yes, this is terrific, thanks Jesse! No additional suggestions from me other than what Lindy mentioned about "very" vs. "verify". 👍

**Comment by lieut-data** [comment]
Thanks, @lindalumitchell & @lindy65. I'm guessing there's something about the `deno` that I need to do to merge? I'm uncertain how to interpret the CI failure.

**Comment by lindalumitchell** [comment]
Ah yes, it's failing to validate on `"customFields.Team Ownership" with unexpected value of "MS Teams"` because the Team Ownership list is out of date. 

Opened https://github.com/mattermost/mattermost-test-management/pull/169 to update that list. Since we're into the weekend for many folks such as Saturn, to avoid waiting until next week would you be willing to review that PR, @lieut-data ? Unconventional, but these are very low-risk changes.

**Comment by lieut-data** [comment]
@lindalumitchell, reviewed, but alas I think a code owner is required on that repo.

**Comment by lindalumitchell** [comment]
Ah you're right. Thanks for reviewing it anyway, and I'm out for two weeks starting Monday, but next week after other folks are back and can get that reviewed and merged in, I think this one should be able to pass validation and be merged as well.

File: data/test-cases/plugins/ms-teams-sync/bot-messages/connection-prompt/MM-TXXX0.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+Disconnect from MS Teams, if necessary, via `/msteams disconnect`.
+
+**Step 2**
+
+Send a direct message to a remote user (e.g. `@msteams_user`)
```
Comment (saturninoabril) [inline data/test-cases/plugins/ms-teams-sync/bot-messages/connection-prompt/MM-TXXX0.md:42]: Multiple "Step #n" in Zephyr looks like in the screenshot, with each "Step" having "test step or steps", test data (optional) and expected result (required). A bit confusing but in general, it's a sub-test case (if you will). Each step has its own test result and should be mapped properly between test automation and Zephyr.
Screenshot 2024-04-01 at 1 04 32 PM

With that, the step to disconnect from MS teams can be combined to the actual step (`Step #2`) or could be as `Precondition`.

```suggestion
**Precondition**

Disconnect from MS Teams, if necessary, via `/msteams disconnect`.

**Step 1**

Send a direct message to a remote user (e.g. `@msteams_user`)
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @lieut-data! Overall looks good except for test structure.

For codeowner, I think I'll be removing it from this repo for this has been stable for a while and very much open to everyone. (will address separately)

**Review by lindy65 — APPROVED** [review]
Thanks @lieut-data - I see Saturn has helped with comments about test structure so I'll go ahead and approve this PR as you can request final review from him after addressing requested changes 👍

**Comment by saturninoabril** [comment]
@lieut-data Let me know if you need anything so we can merge as soon as possible.

File: data/test-cases/plugins/ms-teams-sync/bot-messages/connection-prompt/MM-TXXX0.md (hunk 1 above)
Comment (lieut-data) [reply data/test-cases/plugins/ms-teams-sync/bot-messages/connection-prompt/MM-TXXX0.md:42]: Thanks, @saturninoabril! I've updated as recommended -- let me know if that works?

**Review by saturninoabril — APPROVED** [review]
Thanks @lieut-data, LGTM.

**Pull request merged**